server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1277,6 +1272,13 @@ public String toString() {
         Property.InternalIndex
     );
 
+    public static final Setting<String> INDEX_DOWNSAMPLE_INTERVAL = Setting.simpleString(
+        INDEX_DOWNSAMPLE_INTERVAL_KEY,
+        "",
+        Property.IndexScope,
+        Property.InternalIndex
```

Comment:
I think this should be a private setting as well? Only the downsample api should be allowed to set it.